src/Plugin/views/field/CustomEntityField.php

```diff
@@ -335,4 +334,40 @@ protected function getDelta($id) {
     return 0;
   }
 
+  /**
+   * Check if date field should be converted to UTC or not.
+   *
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $definition
+   *   The field definition.
+   * @param string $date_value
+   *   The date value.
+   *
+   * @return string
+   *   The converted value.
+   */
+  public function convertToUtc(FieldDefinitionInterface $definition, $date_value) {
+    $datetime_format = $definition->getSetting('datetime_type') === DateTimeItem::DATETIME_TYPE_DATE ? DateTimeItemInterface::DATE_STORAGE_FORMAT : DateTimeItemInterface::DATETIME_STORAGE_FORMAT;
+    $default_timezone = date_default_timezone_get();
```

Comment:
This line is the difference from PR #435 for the 3.x branch

Reply:
If using the global configuration then if the per-user timezone setting is allowed, and if the user chooses a different timezone, then could have a discrepancy. Drupal manages this and gives the "right" timezone returned by  date_default_timezone_get();